### PR TITLE
Check if category is associated to the current shop

### PR DESCRIPTION
**Pull request by simondaigre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix check if category is associated to the shop
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | Try to access to a category via its URL. <br>**Before :** Category shows without PR (NOK)<br>**After :** 404 with this PR (OK)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17408)

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
@simondaigre Need fix on PrettyCI before validation ;)

**Comment by simondaigre** [comment]
> @simondaigre Need fix on PrettyCI before validation ;)

Errors thrown on PrettyCI doesn't seems related to my commit ?

**Comment by Progi1984** [comment]
@simondaigre May be an update of Pretty CI ? Could you fix it ?

**Comment by simondaigre** [comment]
@Progi1984 how to do this ?

**Comment by Progi1984** [comment]
@simondaigre Execute `vendor/bin/php-cs-fixer fix --diff --dry-run` at the root of your project and fix each bug. Thanks

**Comment by simondaigre** [comment]
Sure ? Issue is related to PR #17233, not mine.

**Comment by Hherreros** [comment]
@simondaigre It looks like they added an incorrect commit to Develop. I wrote to the developer who added it, i hope they update it soon.

**Comment by Progi1984** [comment]
Hello @simondaigre , would you mind to rebase your pull request for removing this PrettyCI error ? Thanks

**Review by matthieu-rolland — DISMISSED** [review]
lgtm

**Review by PierreRambaud — DISMISSED** [review]

**Comment by simondaigre** [comment]
@Progi1984 could you do it ? i'm not an expert with Git

**Comment by NeOMakinG** [comment]
> @Progi1984 could you do it ? i'm not an expert with Git

Hey, I suggest you to try it, this is not that hard on such a PR

Add a remote upstream to your local git : git remote add upstream git@github.com:PrestaShop/PrestaShop.git

or use http url, depends of your use
then :
git checkout develop
git pull upstream develop
git checkout patch-1
git rebase develop

If you have some conflict, resolve them and git add the files
then 
git rebase --continue

Same thing with conflicts, then when it's fine

git push -f

Hope it helps

**Comment by simondaigre** [comment]
@NeOMakinG Done, thanks 👍

**Review by NeOMakinG — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
Hi @simondaigre Can you provide us more infos on "How to test" please? Is this PR linked to a bug ticket?

**Comment by simondaigre** [comment]
Hi @SD1982 

I didnt create an issue for this bug.

To reproduce it:
1- Create a PS instance with two shops
2- Create a category and restrict it to one of the available shop
3- Go to another shop and try the category url, it works (but it shouldnt).

tldr: In multishop context, categories list are available even if the category is not enabled for the shop.

This PR is just a copy and paste from 1.6 version, I don't know why this check was removed.

**Comment by PierreRambaud** [comment]
Hi @simondaigre ,

An issue is mandatory in our process. I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!
